sharding-orchestration/sharding-orchestration-core-new/src/main/java/org/apache/shardingsphere/orchestration/temp/internal/registry/ShardingOrchestrationFacade.java

```diff
@@ -119,26 +120,24 @@ public void close() {
         }
     }
     
-    private InstanceConfiguration getConfigurationByOrchestrationType(final Map<String, InstanceConfiguration> map, final String type) {
-        if (null == map || null == type) {
-            return null;
+    private Optional<String> getInstanceNameByOrchestrationType(final Map<String, InstanceConfiguration> map, final String type) {
+        if (null == map || 0 == map.size() || null == type) {
+            return Optional.absent();
         }
-        for (Entry<String, InstanceConfiguration> entry: map.entrySet()) {
-            InstanceConfiguration configuration = entry.getValue();
-            if (contains(configuration.getOrchestrationType(), type)) {
-                return configuration;
+        for (Entry<String, InstanceConfiguration> each : map.entrySet()) {
+            if (contains(each.getValue().getOrchestrationType(), type)) {
+                return Optional.of(each.getKey());
             }
         }
-        return null;
+        return Optional.absent();
     }
     
     private boolean contains(final String collection, final String element) {
         if (Strings.isNullOrEmpty(collection)) {
             return false;
         }
-        Iterator<String> iterator = Splitter.on(",").split(collection).iterator();
-        while (iterator.hasNext()) {
-            if (element.equals(iterator.next())) {
+        for (String each : Splitter.on(",").split(collection)) {
```

Comment:
I think this should `Splitter.on(",").split(collection).trim()`, user maybe config the orchestration type with blank space.